app/controllers/my-projects/project/hearing/add.js

```diff
@@ -85,28 +85,30 @@ export default class MyProjectsProjectHearingAddController extends Controller {
     const { dispositions } = this.model;
     const allActions = this.get('allActions');
 
-    try {
-      // if user is submitting ONE hearing for ALL actions
-      if (allActions) {
-        const allActionsDispHearingLocation = this.get('dispositionForAllActions.dcpPublichearinglocation');
-        const allActionsDispHearingDate = this.get('dispositionForAllActions.dcpDateofpublichearing');
-
-        // iterate through each disposition on the modal
-        // set hearing location and date on each one with the single value that the user input
-        // this single value is saved on dispositionForAllActions object
-        dispositions.forEach(function(disposition) {
-          disposition.set('dcpPublichearinglocation', allActionsDispHearingLocation);
-          disposition.set('dcpDateofpublichearing', allActionsDispHearingDate);
-        });
-      }
-      await dispositions.save();
-
-      this.set('hearingSubmitted', false);
-      this.set('checkIfMissing', false);
-      this.set('modalOpen', false);
-      this.transitionToRoute('my-projects.project.hearing.done');
-    } catch (e) {
-      alert('Sorry about that, unable to connect to server, can you try again?'); // eslint-disable-line
+    // if user is submitting ONE hearing for ALL actions
+    if (allActions) {
+      const allActionsDispHearingLocation = this.get('dispositionForAllActions.dcpPublichearinglocation');
+      const allActionsDispHearingDate = this.get('dispositionForAllActions.dcpDateofpublichearing');
+
+      // iterate through each disposition on the modal
+      // set hearing location and date on each one with the single value that the user input
+      // this single value is saved on dispositionForAllActions object
+      dispositions.forEach(function(disposition) {
+        disposition.set('dcpPublichearinglocation', allActionsDispHearingLocation);
+        disposition.set('dcpDateofpublichearing', allActionsDispHearingDate);
+      });
     }
+
+    const self = this;
+
+    await dispositions.save().then(function() {
```

Comment:
I would use an arrow function in line 104 here: `.then(() => {`... the reason being that with arrow functions, we no longer need to use `const self = this` — you can just use `this` inside the arrow function because it inherits the outer scope's `this`. 

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

Reply:
Also, because you're `await`ing the promise returned from `dispositions.save()`, you don't need to use `.then` — this is redundant because `await` implicitly "thens" the promise.

Check it out: https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await